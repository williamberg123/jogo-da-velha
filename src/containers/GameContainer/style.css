.GameContainer {
    max-width: 390px;
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    grid-gap: 5px;
    margin: 30px auto;
    padding: 0 10px;
}