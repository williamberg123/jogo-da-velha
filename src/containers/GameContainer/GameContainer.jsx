import React from 'react';

import PropTypes from 'prop-types';

import Square from '../../components/Square/Square';

import './style.css';

export default function GameContainer({ handleClick }) {
    const htmlRendering = Array(9).fill().map((item, index) => <Square index={index} funcHandleClick={(e) => handleClick(e, index)} key={`square${index}`} />);

    return (
        <div className="GameContainer">
            {htmlRendering}
        </div>
    );
}

GameContainer.propTypes = {
    handleClick: PropTypes.func.isRequired
};
