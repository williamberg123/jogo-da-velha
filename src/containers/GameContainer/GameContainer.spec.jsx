import React from 'react';
import { screen, render, fireEvent } from '@testing-library/react';
import GameContainer from './GameContainer';

describe('<GameContainer />', () => {
    it('should render element correctly', () => {
        const fn = jest.fn();
        const {debug} = render(<GameContainer handleClick={fn} />);
        // debug();

        expect(screen.getAllByTestId('Square')).toHaveLength(9);
    });

    it('should call function when Square is clicked', () => {
        const fn = jest.fn();
        render(<GameContainer handleClick={fn} />);

        fireEvent.click(screen.getAllByTestId('Square')[0]);

        expect(fn).toBeCalledTimes(1);
    });

    it('should match snapshot', () => {
        const fn = jest.fn();
        render(<GameContainer handleClick={fn} />);

        expect(screen.getAllByTestId('Square')[0].parentElement).toMatchSnapshot()
    });
});
