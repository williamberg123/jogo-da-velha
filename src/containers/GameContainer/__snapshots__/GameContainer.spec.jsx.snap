// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameContainer /> should match snapshot 1`] = `
<div
  class="GameContainer"
>
  <div
    class="Square Square0"
    data-testid="Square"
  />
  <div
    class="Square Square1"
    data-testid="Square"
  />
  <div
    class="Square Square2"
    data-testid="Square"
  />
  <div
    class="Square Square3"
    data-testid="Square"
  />
  <div
    class="Square Square4"
    data-testid="Square"
  />
  <div
    class="Square Square5"
    data-testid="Square"
  />
  <div
    class="Square Square6"
    data-testid="Square"
  />
  <div
    class="Square Square7"
    data-testid="Square"
  />
  <div
    class="Square Square8"
    data-testid="Square"
  />
</div>
`;
